src/value.c

```diff
@@ -371,58 +371,59 @@ SIValue SIValue_Divide(const SIValue a, const SIValue b) {
+	uint notEqualCounter = 0;   // Counter for the amount of false (compare(a,b) !=0) comparisons.
+	uint nullCounter = 0;       // Counter for the amount of null comparison.
+	// notEqual holds the first false (result != 0) comparison result between two values from the same type, which are not equal.
 	int notEqual = 0;
 
 	// Go over the common range for both arrays.
-	for(uint i = 0; i < minLengh; i++) {
+	for(uint i = 0; i < minLength; i++) {
 		SIValue aValue = SIArray_Get(arrayA, i);
 		SIValue bValue = SIArray_Get(arrayB, i);
-		int compareResult = SIValue_Compare(aValue, bValue);
-		switch(compareResult) {
-		case 0:
-			// we have a matching value
-			break;
-		case COMPARED_NULL:
-			// there was a null comparison
-			nullCompare++;  // update null comparison counter
-			allDisjoint++;  // null comparison is also a disjoint comparison
-			break;
-		case DISJOINT:
-			allDisjoint++;  // there was a disjoint comparison
-			if(notEqual == 0) notEqual =
-					compareResult; // if there wasn't any false comparison, update the false comparison to disjoint value
-			break;
-		default:
-			// if comparison is not 0 (a != b), set the first value
+		// Current comparison special cases indication variable.
+		int currentDisjointOrNull;
+		int compareResult = SIValue_Compare(aValue, bValue, &currentDisjointOrNull);
+		// In case of special case such null or disjoint comparison.
+		if(currentDisjointOrNull) {
+			if(currentDisjointOrNull == COMPARED_NULL) nullCounter++;   // Update null comparison counter.
+			// Null or disjoint comparison is also a false comparison, so increase the number of false comparisons in one.
+			notEqualCounter++;
+		} else if(compareResult != 0) {
+			// In the normal false comparison case, update false comparison counter.
+			notEqualCounter++;
+			// Set the first difference value, if not set before.
 			if(notEqual == 0) notEqual = compareResult;
-			break;
 		}
+		// Note: In the case of compareResult = 0, there is nothing to be done.
+	}
+	// If all the elements in the shared range yielded false comparisons.
+	if(notEqualCounter == minLength && notEqualCounter > nullCounter) return notEqual;
+	// If there was a null comperison on non disjoint arrays.
+	if(nullCounter) {
+		if(disjointOrNull) *disjointOrNull = COMPARED_NULL;
+		return notEqual;
 	}
-	// if all the elements in the shared range are from disjoint types return DISJOINT array
-	if(allDisjoint == minLengh && allDisjoint > nullCompare) return DISJOINT;
-	// if there was a null comperison on non disjoint arrays
-	if(nullCompare) return COMPARED_NULL;
-	// if there was a diffrence in some member
+	// If there was a difference in some member, without any null compare.
 	if(notEqual) return notEqual;
-	// if all elemnts are equal and length are equal so arrays are equal, otherwise return lengh diff
+	// If all elemnts are equal and length are equal so arrays are equal, otherwise return length diff.
```

Comment:
Comment is misleading it describes a condition: `If ... otherwise..`  but we're simply returning here.

Reply:
fixed to more informative comment